nncf/experimental/common/pruning/operations.py

```diff
@@ -0,0 +1,700 @@
+            input_tensors_shapes = [x.tensor_shape for x in graph.get_input_edges(node)]
+            assert len(input_tensors_shapes) == 1
+            new_order = list(range(len(input_tensors_shapes[0])))
+            dim0 = node.layer_attributes.dim0
+            dim1 = node.layer_attributes.dim1
+            new_order[dim1], new_order[dim0] = new_order[dim0], new_order[dim1]
+        elif isinstance(node.layer_attributes, PermuteLayerAttributes):
+            new_order = node.layer_attributes.permutation
+
+        idx_map = [(old_idx, new_idx) for new_idx, old_idx in enumerate(new_order) if
+                   old_idx in input_mask.dim_groups_map]
+        output_mask = PropagationMask(
+            dim_groups_map={
+                new_idx: input_mask.dim_groups_map[old_idx] for old_idx, new_idx in idx_map
+            }
+        )
+
+        node.data['output_mask'] = output_mask
+
+class StopMaskForwardPruningOp(BasePruningOp):
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        input_masks = get_input_masks(node, graph)
+        for m in input_masks:
+            if m:
+                m.invalidate_groups()
+        node.data['output_mask'] = None
+
+class ExpandAsPruningOp(BasePruningOp):
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        input_edges = graph.get_input_edges(node)
+        assert len(input_edges) == 2, 'expand should always have 2 inputs'
+        input_to_expand = input_edges[0].from_node
+        mask = input_to_expand.data.get('output_mask')
+        propagated_mask = None
+        if mask is not None:
+            nncf_logger.warning(f"expand_as is applied to the node with propagation mask. Currently, it's not supported "
+                                "and mask is invalidated. node_name={node.node_name}")
+            mask.invalidate_groups()
+        input_to_get_shape = input_edges[1].from_node
+        mask: PropagationMask = input_to_get_shape.data.get('output_mask')
+        if mask is not None:
+            target_shape = input_edges[1].tensor_shape
+            source_shape = input_edges[0].tensor_shape
+            for dim, groups in mask.dim_groups_map.items():
+                if target_shape[dim] == source_shape[dim]:
+                    nncf_logger.warning(f"expand_as takes the shape from the node with propagation mask and pruning "
+                                        "dimension in the mask matches the dimension in the expanded input. Currently, "
+                                        "it's not supported and mask is invalidated. node_name={node.node_name}")
+                    for group in groups:
+                        group.invalidate()
+            propagated_mask = mask
+        node.data['output_mask'] = propagated_mask
```

Comment:
Discussed locally. NGraph implementation has the same limitation. It will work only if attention mask is always filled with zeros, which is the case for inference. 
cc: @AlexKoff88